src/app/dialogs/node-versions/node-versions.dialog.ts

```diff
@@ -36,12 +42,32 @@ import { Store } from '@ngrx/store';
 })
 export class NodeVersionsDialogComponent {
   node: MinimalNodeEntryEntity;
+  file: File;
+  typeList = true;
 
-  constructor(@Inject(MAT_DIALOG_DATA) data: any, private store: Store<any>) {
-    this.node = data.node;
+  /** Emitted when a version is restored or deleted. */
+  @Output()
+  refreshEvent: EventEmitter<Node> = new EventEmitter<Node>();
+
+  constructor(
+    @Inject(MAT_DIALOG_DATA) data: any,
+    private store: Store<any>,
+    private dialogRef: MatDialogRef<NodeVersionsDialogComponent>
+  ) {
+    this.node = data.node ? data.node : undefined;
+    this.file = data.file ? data.file : undefined;
+    this.typeList = data.typeList != undefined ? data.typelist : true;
```

Comment:
`typeList` with value boolean maybe should have a better name like  `isTypeList` or `isList`?

Reply:
modified with the requests